Add password strength meter

veeara282 opened the pull request:
Closes #247

Add a password strength meter to the "New password" fields in `ChangePasswordModal` and `ResetPasswordModal` using a `PasswordStrengthModalInput`, which composes `ModalInput`. This component will always display validation text, even when the password is valid.

Password strength indicator colors:
- Weak/invalid: `#f05452` (tangerine)
- Okay: `#43c4d9` (aqua)
- Good: `#388ec5`
- Great: `#2c58b1` (bluejay)

TODO:
- [x] Allow `{Reset,Change}PasswordModal` to read the validation result (true/false) of the `PasswordStrengthModalInput`
- [x] Color shouldn't be red when the password is valid

veeara282 committed:
Add password strength meter

veeara282 committed:
Please the eslint god

veeara282 committed:
remove ignored props

veeara282 committed:
Add to ResetPasswordModal

veeara282 committed:
Add component description

colbymorrison committed:
update test

aidan-fitz commented:
We should also consider adding some default and personalized strings to the `user_inputs` argument.

> The optional `user_inputs` argument is an array of strings that zxcvbn will treat as an extra dictionary. This can be whatever list of strings you like, but is meant for user inputs from other fields of the form, like name and email. That way a password that includes a user's personal information can be heavily penalized. This list is also good for site-specific vocabulary — Acme Brick Co. might want to include ['acme', 'brick', 'acmebrick', etc].

veeara282 committed:
Add colors to password strength indicator

Vary the color of the validation message depending on the score.
- Weak/invalid: `#f05452` (tangerine)
- Okay: `#43c4d9` (aqua)
- Good: `#388ec5`
- Great: `#2c58b1` (bluejay)

veeara282 committed:
format

veeara282 committed:
Have PSMI update parent's passwordIsValid state

veeara282 committed:
Don't use weak password in test

ibeckermayer reviewed (commented):
Ignore, I'm just clearing out old review requests in my GitHub account.
